sklearn/cluster/_agglomerative.py

```diff
@@ -911,6 +938,7 @@ def fit(self, X, y=None):
         self : object
             Returns the fitted instance.
         """
+        self._validate_params()
```

Comment:
Remove the following manual checks:

l.933

```python
        if self.n_clusters is not None and self.n_clusters <= 0:
            raise ValueError(
                "n_clusters should be an integer greater than 0. %s was provided."
                % str(self.n_clusters)
            )
```

l.957

```python
        if self.linkage not in _TREE_BUILDERS:
            raise ValueError(
                "Unknown linkage type %s. Valid options are %s"
                % (self.linkage, _TREE_BUILDERS.keys())
            )
```

Reply:
Done.